src/features/hooks/analytic/use-analytic-events/index.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react'
+import { isMobile } from 'react-device-detect'
+import { useAnalyticData } from 'features/hooks/analytic/use-analytic-data'
+
+type AnalyticsTypes = 'ce_virtual_signup_form' | 'ce_real_account_signup_form'
+
+type AnalyticsEventsTypes =
+    | 'open'
+    | 'started'
+    | 'email_confirmation_sent'
+    | 'email'
+    | 'google'
+    | 'facebook'
+    | 'apple'
+
+export const useAnalyticsEvents = (event: AnalyticsTypes) => {
+    const { form_source, form_name } = useAnalyticData()
+
+    const analytic_events = {
+        [event]: useCallback((action: AnalyticsEventsTypes) => {
+            window.rudderanalytics.track(
+                event,
+                { action, form_source, form_name },
+                // for demo
+                () => {
+                    console.log(`${isMobile ? `${action} on mobile` : action}`, `${event} happened`)
```

Comment:
Do we need the console?
```suggestion
                    console.log(`${isMobile ? `${action} on mobile` : action}`, `${event} happened`)
```

Reply:
need for testing purpose